File: csharp/Platform.Collections.Tests/StringTests.cs
using Xunit;

namespace Platform.Collections.Tests
{
    public static class StringTests
    {
        [Fact]
        public static void CapitalizeFirstLetterTest()
        {
            Assert.Equal("Hello", "hello".CapitalizeFirstLetter());
            Assert.Equal("Hello", "Hello".CapitalizeFirstLetter());
            Assert.Equal("  Hello", "  hello".CapitalizeFirstLetter());
        }

        [Fact]
        public static void TrimSingleTest()
        {
            Assert.Equal("", "'".TrimSingle('\''));
            Assert.Equal("", "''".TrimSingle('\''));
            Assert.Equal("hello", "'hello'".TrimSingle('\''));
            Assert.Equal("hello", "hello'".TrimSingle('\''));
            Assert.Equal("hello", "'hello".TrimSingle('\''));
        }

        [Fact]
        public static void EqualsIgnoreCaseTest()
        {
            Assert.True("Hello".EqualsIgnoreCase("hello"));
            Assert.True("HELLO".EqualsIgnoreCase("hello"));
            Assert.True("hello".EqualsIgnoreCase("HELLO"));
            Assert.True("Hello World".EqualsIgnoreCase("HELLO WORLD"));
            Assert.True("".EqualsIgnoreCase(""));
            Assert.False("Hello".EqualsIgnoreCase("World"));
            Assert.False("Hello".EqualsIgnoreCase("Hell"));
            Assert.False("Hello".EqualsIgnoreCase(null!));
            Assert.False(((string?)null).EqualsIgnoreCase("Hello"));
            Assert.True(((string?)null).EqualsIgnoreCase(null));
        }
    }
}

File: cpp/Platform.Collections.Tests/StringTests.cpp
namespace Platform::Collections::Tests
{
    TEST(StringTests, IsWhiteSpace)
    {
        using namespace std::string_literals;

        std::locale::global(std::locale(""));
        ASSERT_TRUE(IsWhiteSpace(""s));
        ASSERT_TRUE(IsWhiteSpace(" "s));
        ASSERT_TRUE(IsWhiteSpace("\n\v\f\r"s));
        ASSERT_TRUE(IsWhiteSpace(u"\u2003"s));
    }

    TEST(StringTests, CapitalizeFirstLetterTest)
    {
        using namespace std::string_literals;

        std::locale::global(std::locale(""));
        ASSERT_EQ(u"Привет"s, CapitalizeFirstLetter(u"привет"s));
        ASSERT_EQ("Hello", CapitalizeFirstLetter("hello"s));
        ASSERT_EQ("Hello", CapitalizeFirstLetter("Hello"s));
        ASSERT_EQ("  Hello", CapitalizeFirstLetter("  hello"s));
    }

    TEST(StringTests, TrimSingleTest)
    {
        using namespace std::string_literals;

        ASSERT_EQ("", TrimSingle("'"s, '\''));
        ASSERT_EQ("", TrimSingle("''"s, '\''));
        ASSERT_EQ("hello", TrimSingle("'hello'"s, '\''));
        ASSERT_EQ("hello", TrimSingle("hello'"s, '\''));
        ASSERT_EQ("hello", TrimSingle("'hello"s, '\''));
    }
}
